ocw/lib/aks.py

```diff
@@ -0,0 +1,70 @@
+import os
+import kubernetes
+from azure.identity import ClientSecretCredential
+from azure.mgmt.containerservice import ContainerServiceClient
+from ocw.lib.provider import Provider
+from ocw.lib.k8s import clean_jobs
+
+
+class AKS(Provider):
+    __instances = {}
+    default_region: str = 'eu-central-1'
+
+    def __new__(cls, vault_namespace):
+        if vault_namespace not in AKS.__instances:
+            AKS.__instances[vault_namespace] = object.__new__(cls)
+            AKS.__container_mgmt_client = None
+            AKS.__sp_credentials = None
+            AKS.__kubectl_client = {}
+        return AKS.__instances[vault_namespace]
+
+    def subscription(self) -> str:
+        return self.get_data('subscription_id')
+
+    def resource_group_name(self) -> str:
+        return self.get_data('resource_group_name')
```

Comment:
AFAIK this files does not have `'resource_group_name'` at least the one which we have in prod certainly does not have it . Also this is single file which is global for single account and this field should be specific for certain single cluster

Reply:
I think it won't work we need to find another way to get resource group name for AKS cluster

Reply:
done